### PR TITLE
fix(inventory): exclude 0.0.0.0 from management port

### PR DESCRIPTION
**Pull request by stonebuzz**
Alternative to https://github.com/glpi-project/glpi/pull/13591

Exclude ```0.0.0.0``` from ``` management port```

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://github.com/glpi-project/glpi/issues/13523#issuecomment-1347896226

### CODE REVIEW

File: src/Inventory/Asset/NetworkEquipment.php [hunk 1]
```diff
@@ -111,7 +114,12 @@ public function prepare(): array
 
                //add internal port(s)
                 foreach ($device->ips as $ip) {
-                    if ($ip != '127.0.0.1' && $ip != '::1' && !in_array($ip, $port->ipaddress)) {
+                    if (
+                        $ip != '127.0.0.1'
```
[the remark this answers is not in the document]
Comment (stonebuzz) [reply src/Inventory/Asset/NetworkEquipment.php:118]: ```127.0.0.1``` seems to exist already

```sql
+----+------+-----------+-----------+---------+----------+---------------+
| id | type | name      | value     | comment | date_mod | date_creation |
+----+------+-----------+-----------+---------+----------+---------------+
| 69 |    1 | empty IP  |           | NULL    | NULL     | NULL          |
| 70 |    1 | zero IP   | 0.0.0.0   | NULL    | NULL     | NULL          |
| 71 |    1 | localhost | 127.0.0.1 | NULL    | NULL     | NULL          |
+----+------+-----------+-----------+---------+----------+---------------+
```

[the remark this answers is not in the document]
Comment (stonebuzz) [reply src/Inventory/Asset/NetworkEquipment.php:118]: I'm looking to add ```::1```

[the remark this answers is not in the document]
Comment (trasher) [reply src/Inventory/Asset/NetworkEquipment.php:118]: :+1: 
than condition can be simplified here :)

**Comment by periodistacg** [comment]
Hi
I am trying to fix this, but I do not understand then change on line 114.
Is it correct?
image

**Comment by stonebuzz** [comment]
Please use patch -> https://github.com/glpi-project/glpi/pull/13594.diff

or this build -> https://github.com/stonebuzz/glpi/actions/runs/3686755286

But if you are not comfortable with ```git``` notions it will be complicated to test

**Comment by periodistacg** [comment]
I have changed both files (backlist and networkequipment) and it works fine. The 0.0.0.0 is not imported now

File: src/Inventory/Asset/NetworkEquipment.php [hunk 2]
```diff
@@ -36,6 +36,8 @@
 
 namespace Glpi\Inventory\Asset;
 
+use Blacklist;
+use BlacklistedMailContent;
```
Comment (trasher) [inline src/Inventory/Asset/NetworkEquipment.php:40]: Does not seems useful
```suggestion
```

**Review by trasher — APPROVED** [review]

**Pull request merged**